include/drivers/i2c.h

```diff
@@ -252,6 +255,31 @@ static inline int z_impl_i2c_transfer(struct device *dev,
 	return api->transfer(dev, msgs, num_msgs, addr);
 }
 
+/**
+ * @brief Recover the I2C bus
+ *
+ * Attempt to recover the I2C bus.
+ *
```

Comment:
Please document the procedure required in some SoCs to switch the pins from I2C to GPIO and back.